### PR TITLE
🐛 fix: auto test connot generate report

### PR DESCRIPTION
**Pull request by jin860315**
- 驗證有遇到錯誤，可正常完成測試並且產出報告：
https://coolbitx-technology.github.io/coolwallet-sdk/all-services/36/
- 驗證所有幣種可正常執行測試並產出報告：
https://coolbitx-technology.github.io/coolwallet-sdk/all-services/37/
----

*Checklist:*

- README.md includes:
  - [ ] The details of signing data and transaction data, this includes parameters.
  - [ ] Official docs or white paper.
  - [ ] Website or api can query assets and broadcast transaction.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

 
 **PR Summary by Typo**
------------

 **Summary:**
This pull request updates testing scripts and configurations for multiple packages, enabling continuous integration with improved test reporting and reliability.

**Key Points:**
1. Added `--no-bail` flag to `ci-test-report` scripts in several packages.
2. Changed Jest testing environments and reporters for better compatibility.
3. Updated build scripts to use TypeScript and Babel.
4. Modified handling-report.sh script for JUnit XML processing. 

 <h6>To turn off PR summary, please visit <a href="https://app.typoapp.io/settings/dev-analytics/notification?tab=codeHealth">Notification settings</a>.</h6>

### CODE REVIEW

File: .github/workflows/test-and-generate-report.yml [hunk 1]
```diff
@@ -85,7 +85,7 @@ jobs:
             echo "Test file found for = $modified_coin"
             cd ${{ github.workspace }}/sdk/$modified_coin
             npm install
-            npm run ci-test-report
+            npm run ci-test-report || true
```
Comment (jin860315) [inline .github/workflows/test-and-generate-report.yml:88]: 指令加上 `|| true` 當遇到 test failed 時還是會繼續跑下一個幣種的測試

File: .github/workflows/test-and-generate-report.yml [hunk 2]
```diff
@@ -101,7 +101,7 @@ jobs:
       with:
         name: result
         path: |
-          ${{ github.workspace }}/scripts/reports/junit/
+          ${{ github.workspace }}/sdk/scripts/reports/junit/
```
Comment (jin860315) [inline .github/workflows/test-and-generate-report.yml:104]: 多增加一層`sdk`資料夾才找得到檔案

File: packages/coin-ada/package.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
   "scripts": {
     "test": "NODE_ENV=development jest --runInBand",
     "ci-test": "NODE_ENV=production jest --runInBand",
-    "ci-test-report": "NODE_ENV=development jest --runInBand --reporters=default --reporters=jest-junit --testResultsProcessor=jest-junit",
+    "ci-test-report": "NODE_ENV=production jest --no-bail --runInBand --reporters=default --reporters=jest-junit --testResultsProcessor=jest-junit",
```
Comment (jin860315) [inline packages/coin-ada/package.json:11]: 調整 NODE_ENV 為 production，跟 ci-test 一樣

File: scripts/handing-report.sh [hunk 4]
```diff
@@ -5,7 +5,7 @@ mkdir -p scripts/reports/junit
 for dir in packages/coin-* packages/core; do
   if [ -d "$dir" ] && [ -f "$dir/junit.xml" ]; then
     coin_name=$(basename "$dir")
-    sed 's/&quot;/"/g' "$dir/junit.xml" >"scripts/reports/junit/${coin_name}-junit.xml" || echo "Failed to process $dir/junit.xml"
+    sed 's|<failure>|<failure message="Error">|g' "$dir/junit.xml" >"scripts/reports/junit/${coin_name}-junit.xml" || echo "Failed to process $dir/junit.xml"
```
Comment (jin860315) [inline scripts/handing-report.sh:8]: 在 xml 檔裡面多增加 failure message 讓 allure report 會有 error trace

**Review by tp26610 — APPROVED** [review]

**Pull request merged**